tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -151,101 +161,184 @@ int main(int argc, char** argv) {
-        }
+            for (int i = 0; i < NUM_KERNEL_DDR; i++) {
+                OCL_CHECK(err, err = q.enqueueWriteBuffer(input_buffer[i], CL_TRUE, 0, vector_size_bytes,
+                                                          input_host.data(), nullptr, nullptr));
+                OCL_CHECK(err, err = q.finish());
+            }
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            OCL_CHECK(err, err = krnls[i].setArg(0, input_buffer[i]));
-            OCL_CHECK(err, err = krnls[i].setArg(1, output_buffer[i]));
-            OCL_CHECK(err, err = krnls[i].setArg(2, DATA_SIZE));
-            OCL_CHECK(err, err = krnls[i].setArg(3, reps));
+            std::chrono::high_resolution_clock::time_point timeStart;
+            std::chrono::high_resolution_clock::time_point timeEnd;
+
+            timeStart = std::chrono::high_resolution_clock::now();
+
+            for (int i = 0; i < NUM_KERNEL_DDR; i++) {
+                OCL_CHECK(err, err = q.enqueueTask(krnls[i]));
+            }
+            q.finish();
+            timeEnd = std::chrono::high_resolution_clock::now();
+
+            for (int i = 0; i < NUM_KERNEL_DDR; i++) {
+                OCL_CHECK(err, err = q.enqueueReadBuffer(output_buffer[i], CL_TRUE, 0, vector_size_bytes,
+                                                         output_host[i].data(), nullptr, nullptr));
+                OCL_CHECK(err, err = q.finish());
+            }
+
+            // check
+            for (int i = 0; i < NUM_KERNEL_DDR; i++) {
+                for (uint32_t j = 0; j < DATA_SIZE; j++) {
+                    if (output_host[i][j] != input_host[j]) {
+                        std::cout << "ERROR : kernel failed to copy entry " << j << " input " << input_host[j]
+                                  << " output " << output_host[i][j] << std::endl;
+                        return EXIT_FAILURE;
+                    }
+                }
+            }
+
+            double usduration;
+            double dnsduration;
+            double dsduration;
+            double bpersec;
+            double mbpersec;
+
+            usduration =
+                (double)(std::chrono::duration_cast<std::chrono::nanoseconds>(timeEnd - timeStart).count() / reps);
+
+            dnsduration = (double)usduration;
+            dsduration = dnsduration / ((double)1000000000);
+            bpersec = (DATA_SIZE * NUM_KERNEL_DDR) / dsduration;
+            mbpersec = (2 * bpersec) / ((double)1024 * 1024); // For concurrent Read/Write
+
+            if (mbpersec > max_throughput) max_throughput = mbpersec;
         }
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            OCL_CHECK(err, err = q.enqueueWriteBuffer(input_buffer[i], CL_TRUE, 0, vector_size_bytes, input_host.data(),
+        std::cout << "Overall DDRs (total " << NUM_KERNEL_DDR << ") Throughput: " << max_throughput << "MB/s\n";
+    }
+    if (chk_hbm_mem) {
+        for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
```

Comment:
magic?

Reply:
We are trying to run with different buffer sizes(4KB -> 16MB).